### PR TITLE
hello-java-maven example with k8s deployment scripts

### PR DESCRIPTION
**Pull request by renatocaval**
This is an example that uses reactive-lib and maven (but not sbt-reactive-app plugin, obviously) and provides k8s yml files to deploy the services.

The generated docker containers do not have any metadata (labels) and cannot be used with the `rp` tool. The yml files are generated by hand. 

The `rp-application.conf` is also hand-written and it's stripped down to a minimal.

Services uses Akka Cluster Bootstrap for cluster formation and reactive-lib ServiceLocator based on akka-dns. 

TODOs
- [x] add a README file with instructions
- [x] document when to use RBAC
- [x] document what to modify in the yml for production, currently setup for testing in Minikube using Lightbend's Helm chart
- [x] Ingress file for services

### CODE REVIEW

**Comment by ignasi35** [comment]
Hi @renatocaval the filecount on this PR looked weird and I noticed you've pushed some IDE-specific files (`.settings/`, `.project`, `.classpath`, `xyz.iml`, ...) is that intentional?

If those were pushed by mistake I guess the `.gitignore` in the root of the repo is incomplete.

**Comment by renatocaval** [comment]
sure, for the eclipse files. I will remove it and do a force --push

also for the integration-tests

The stream and persistence I don't agree. I think we should be able to show how:
* we can configure Cassandra and Kafka to work in Minikube
* we need to be able to confirm that interservice communication is working

**Comment by renatocaval** [comment]
Removed what I mentioned above. Still, 2894 lines!

**Comment by TimMoore** [comment]
Agree that it's important to demonstrate end-to-end communication, Cassandra and Kafka.

File: kubernetes-deployment/hello-kubernetes-java-mvn/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+.factorypath
```
[the remark this answers is not in the document]
Comment (renatocaval) [reply kubernetes-deployment/hello-kubernetes-java-mvn/.gitignore]: This is being added by the fabric8 maven plugin, if I recall correctly. It doesn't make sense to add on the top-level, IMO.

File: kubernetes-deployment/hello-kubernetes-java-mvn/README.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+hello-stream-v1-0-snapshot-67775f58db-qp2vr   1/1       Running            0          27m
+hello-stream-v1-0-snapshot-67775f58db-w97xm   1/1       Running            0          27m
+hello-v1-0-snapshot-86cdd499c4-2jklj          1/1       Running            0          27m
+hello-v1-0-snapshot-86cdd499c4-6x89c          1/1       Running            0          27m
+hello-v1-0-snapshot-86cdd499c4-v5mhm          1/1       Running            0          27m
+reactive-sandbox-85bf697cc9-wt7zm             1/1       Running            0          1h
+```
+
+Once everyhing is running, you can access the servcices through the external Minikube address.  The service API are exposed on port 80.
+In your browser, open the url: [http://192.168.99.100/api/hello/Joe](http://192.168.99.100/api/hello/Joe).
```
[the remark this answers is not in the document]
Comment (renatocaval) [reply kubernetes-deployment/hello-kubernetes-java-mvn/README.md]: This is the default IP for Minikube and I thought that we could be complicating things by being too zealous. I can add it back.

File: kubernetes-deployment/hello-kubernetes-java-mvn/stream-impl/pom.xml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            <plugin>
+                <groupId>io.fabric8</groupId>
+                <artifactId>docker-maven-plugin</artifactId>
+                <configuration>
+                    <skip>false</skip>
+                    <images>
+                        <image>
+                            <build>
+                                <entryPoint>
+                                    java $JAVA_OPTS -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -XX:MaxRAMFraction=3 -cp '/maven/*' play.core.server.ProdServerStart
```
Comment (renatocaval) [inline kubernetes-deployment/hello-kubernetes-java-mvn/stream-impl/pom.xml]: I just realise that we should not be using experimental flags and try to automatically decide the amount of memory we want to reserve. 

This should be added to the yml file. It must be some well defined values and live in the same file we are defining the available container memory. 

Then one may have different memory settings depending on the environment.

File: kubernetes-deployment/hello-kubernetes-java-mvn/.gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (TimMoore) [reply kubernetes-deployment/hello-kubernetes-java-mvn/.gitignore]: Oh OK. I Googled the file name and only found references to Eclipse adding it.

**Comment by renatocaval** [comment]
fixed conflicts

**Pull request merged**